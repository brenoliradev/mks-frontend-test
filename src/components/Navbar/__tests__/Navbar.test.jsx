import { render, screen } from '@testing-library/react'

import { Navbar } from '..'

describe('<Navbar />', () => {
  it('should render all the Navbar text', () => {
    render(<Navbar />)
  })

  it('should render logo correctly', () => {
    const { container } = render(<Navbar />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    const { container } = render(<Navbar />)

    expect(container.firstChild).toHaveStyle({ 'background-color': '#0f52ba' })
  })
})
