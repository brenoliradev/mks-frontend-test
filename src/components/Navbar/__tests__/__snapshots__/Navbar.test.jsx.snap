// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should render logo correctly 1`] = `
.c0 {
  background-color: #0f52ba;
  color: #fff;
  width: 100%;
  height: 101px;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  grid-area: navbar;
  padding-right: 88px;
  padding-left: 65px;
}

.c1 {
  width: 100%;
  margin: 0 auto;
  max-width: 1800px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  list-style-type: none;
}

.c5 {
  background: #ffffff;
  border-radius: 8px;
  width: 90px;
  height: 45px;
  color: #000;
  padding-left: 15px;
  padding-right: 27px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  font-family: 'Montserrat';
  font-style: normal;
  font-weight: 700;
  font-size: 18px;
  line-height: 22px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.75rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
}

.c3 {
  font-family: 'Montserrat';
  font-style: normal;
  font-weight: 600;
  font-size: 40px;
  line-height: 19px;
  color: fff;
}

.c4 {
  font-family: 'Montserrat';
  font-style: normal;
  font-weight: 300;
  font-size: 20px;
  line-height: 19px;
  padding-top: 0.8rem;
  color: #ffffff;
}

@media (max-width:600px) {
  .c0 {
    padding: 0 1.2rem;
  }
}

<header
  class="c0"
>
  <ul
    class="c1"
  >
    <li
      class="c2"
    >
      <h1
        class="c3"
      >
        MKS
      </h1>
      <span
        class="c4"
      >
        Sistemas
      </span>
    </li>
    <li
      class="c5"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg xmlns=%27http://www.w3.org/2000/svg%27 version=%271.1%27 width=%2719%27 height=%2718%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="A Shopcart Icon"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <span
        class="c6"
      >
        0
      </span>
    </li>
  </ul>
</header>
`;
